### ISSUE TITLE
ApplicationLoopback sample passes AUDCLNT_STREAMFLAGS_AUTOCONVERTPCM to wrong parameter?

### ISSUE THREAD

**Issue by jpark37** [microsoft/Windows-classic-samples#196]
Seems like AUDCLNT_STREAMFLAGS_AUTOCONVERTPCM is being passed to `hnsPeriodicity` instead of `StreamFlags`.

Other questions:

The sample suggests `GetMixFormat` is supposed to work for clients created with `AUDIOCLIENT_ACTIVATION_PARAMS`, but I'm seeing `E_NOTIMPL`. Maybe that's because I'm using 21H1?

The docs say `AUDIOCLIENT_ACTIVATION_PARAMS` needs `20348`, but I don't know where that fits into Windows 10/11 versions. As someone who only has access to public/insider builds, is there a build ready to develop against this API? Someone says this API works as far back as 20H1, but maybe the API isn't fully baked on older Windows 10, and our application shouldn't allow that.

**Comment by vkellerMersive** [comment]
+1

**Comment by F3-L1x** [comment]
Are there any updates on the `GetMixFormat` on Windows 10? (not 20348 == Windows Server 2022)
The Loopback sample itself works.
Thanks in advance for any response.

**Comment by FactoryXCode** [comment]
The comments in the sample are not easily to understand: 
  // Initialize the AudioClient in Shared Mode with the user specified buffer.
  // Note: - Shared Mode is needed when rendering from an audio application or process.
  //       - Exclusive Mode is used when rendering from a hardware endpoint.
  //       - Interface methods that are reffering to audioendpoints, will not work and returns E_NOTIMPL,
  //         for example: GetBufferSize(), IsFormatSupported(), GetDevicePeriod() and GetMixFormat() methods.

The comment does not mention why and how the interface methods are referring to audio endpoints. Also IAudioClient2 and IAudioClient3 are not supported.

In short I think it's an internal Windows affair where a method querying the WAS has some issues.
 
A workaround for IAudioClient missing implementations has been posted by Matthew van Eerde [See](https://matthewvaneerde.wordpress.com/2017/10/17/how-to-negotiate-an-audio-format-for-a-windows-audio-session-api-wasapi-client/) 

Another thing Microsoft should be implement is to return a HResult code when calling ActivateAudioInterfaceAsync with an interface that is not supported.

**Comment by imba-tjd** [comment]
the first issue has been fixed in https://github.com/microsoft/Windows-classic-samples/commit/0b4e48a88ba446f3d87c5f7df12b33a74899a1e1
